### ISSUE TITLE
ERR_HTTP2_PROTOCOL_ERROR and CORS policy

### ISSUE THREAD

**Issue by wings2004s** [jw-12138/r2-uploader#2]
Hello,
While trying to upload the entire folder I have issues.
First few files uploads correctly, but then errors starts appearing.

![image](https://github.com/jw-12138/r2-uploader/assets/163059703/a52a9adf-d253-42b2-bbec-18b0c37a0ebb)

What is also weird, sometimes I get the error that payload is too large even tho I am uploading just ~150mb files
![image](https://github.com/jw-12138/r2-uploader/assets/163059703/858eaef4-470a-4177-b880-dda352ef3d45)

Any suggestions what may go wrong?
thanks for the answer and have a great day/evening/night (depending when are you reading this issue).

**Comment by jw-12138** [comment]
@wings2004s thanks for the feedback, can you please first share some info for me?

1. what deployment of r2-uploader are you using? is it a self-deployed environment or just `https://r2.jw1.dev`?
2. did you modify the code of workers that I provided? if you did, could you please share the code?
3. do you have any proxy or VPN running in the background?

and before you answer these questions, I strongly recommend you to try and setup a custom domain for the workers, a custom domain might work more stable in my sense.

**Comment by wings2004s** [comment]
> @wings2004s thanks for the feedback, can you please first share some info for me?
> 
> 1. what deployment of r2-uploader are you using? is it a self-deployed environment or just `https://r2.jw1.dev`?
> 2. did you modify the code of workers that I provided? if you did, could you please share the code?
> 3. do you have any proxy or VPN running in the background?
> 
> and before you answer these questions, I strongly recommend you to try and setup a custom domain for the workers, a custom domain might work more stable in my sense.

Hello,
thanks for the reply!

1. I did try both ways. The same error happens for both - your hosted website and self-deployed.
2. I put exactly the same code as you suggested in the guide. I also tried some modifications with the help of AI regarding to the CORS issue but none of the solutions worked. I also did try to run that in a few browsers (firstly I had an issue with brave because folder upload is not supported there. Then I tried Edge and chrome (I also tried a incognito modes). None of that worked.
3. I am not using VPN. The only service running on my PC which could break it is a docker..but I just turned it off and the issue still happens.

I was using a custom domain since the beginning. Then I removed it and tried without a custom domain because I though that custom domain can cause the issue. But unfortunately doesn't work with nor without a custom domain :(

**Comment by wings2004s** [comment]
Ok, I am officially stupid.
I do not have VPN installed..but I do have Http Debugger. I forgot that I installed it months ago and it is still running in the background.
Once I stopped the service, it seems to work ❤️ 

Thanks for your help. And thanks for your project. I really like it.
I can only recommend adding in the readme that "folder upload" isn't supported on every browser. I spent 30 minutes modifying the code (I'm vue newbie) to "enable" the button in Brave...until I checked the console where I realised the browser was not supported.
Anyway, once again, I love your project.
Thanks for your work and have a great time!
